tokio/src/io/driver/scheduled_io.rs

```diff
@@ -212,10 +213,7 @@ impl ScheduledIo {
     }
 
     fn wake0(&self, ready: Ready, shutdown: bool) {
-        const NUM_WAKERS: usize = 32;
-
-        let mut wakers: [Option<Waker>; NUM_WAKERS] = Default::default();
-        let mut curr = 0;
+        let mut wakers = WakeList::new();
```

Comment:
Here's a thought: Are we not protecting against panics here in the wrong way? Should we really be panicking and not waking the other wakers just because someone gave us a waker that emitted a panic?

Maybe we should just catch all panics that happen when calling `wake` and ignore them.

Reply:
These wakers are user supplied and could run literally any code in the `wake` call. That said, I would not expect this to ever happen.

Reply:
> Should we really be panicking and not waking the other wakers just because someone gave us a waker that emitted a panic?

Hmm, that's a good point. If one waker panics, failing to wake the others could result in those tasks never being notified.

On the other hand, what's the overhead of adding a `catch_unwind` in this fairly hot loop? Is that worth introducing to handle a case which can only happen if a user-supplied waker is not "well-behaved"?

Reply:
We can still have the destructor call `wake` on the others without a `catch_unwind`.

Reply:
> We can still have the destructor call `wake` on the others without a `catch_unwind`.

A destructor panicing would be even worse, since it could cause a double panic, which would result into an abort

Reply:
Can't user-supplied waker's Drop implementation panic as well? This will cause double panic in the Drop implementation even without waking

Reply:
Yes, if the wakers panic in the destructor, you can get a double abort.

Reply:
It looks like it's absolutely critical to not panic in any of the waker functions. The alternative approach is not to try to fix the erroneous implementation, but to forcefully abort the execution